tests/mocks/Matchers.h

```diff
@@ -0,0 +1,63 @@
+/*
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Library General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Matches.h
+ * Useful matchers.
+ * Copyright (C) 2015 Simon Newton
+ */
+
+#ifndef TESTS_MOCKS_MATCHERS_H_
+#define TESTS_MOCKS_MATCHERS_H_
+
+#include <gmock/gmock.h>
+#include <stdint.h>
+
+// A tuple matcher that checks a message payload.
+class DataMatcher :
+    public testing::MatcherInterface< ::testing::tuple<const void*, unsigned int> > {
+ public:
+  DataMatcher(const uint8_t* expected_data, unsigned int expected_size)
+      : m_expected_data(expected_data),
+        m_expected_size(expected_size) {
+  }
+
+  bool MatchAndExplain(
+      ::testing::tuple<const void*, unsigned int> args,
+      testing::MatchResultListener* listener) const;
+
+  void DescribeTo(::std::ostream* os) const {
+    *os << "matches the data of size " << m_expected_size;
```

Comment:
Same comment, this produces:

    Expected args: are a tuple whose fields (#3, #4) matches the data of size 8